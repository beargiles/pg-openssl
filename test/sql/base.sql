\set ECHO none
BEGIN;
\i sql/openssl.sql
\set ECHO all

--
-- list ciphers.
--
-- Note: this test may fail in future when openssl library is updated.
--
select * from pgx_openssl_list_ciphers();

--
-- list digests.
--
-- Note: this test may fail in future when openssl library is updated.
--
select * from pgx_openssl_list_digests() limit 20;

ROLLBACK;
