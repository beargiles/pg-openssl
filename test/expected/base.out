\set ECHO 0
-- You should write your tests
SELECT openssl('foo', 'bar');
 openssl 
---------
 (foo,bar)
(1 row)

SELECT 'foo' #? 'bar' AS arrowop;
  arrowop  
-----------
 (foo,bar)
(1 row)

CREATE TABLE ab (
    a_field openssl
);
INSERT INTO ab VALUES('foo' #? 'bar');
SELECT (a_field).a, (a_field).b FROM ab;
  a  |  b  
-----+-----
 foo | bar
(1 row)

SELECT (openssl('foo', 'bar')).a;
  a  
-----
 foo
(1 row)

SELECT (openssl('foo', 'bar')).b;
  b  
-----
 bar
(1 row)

SELECT ('foo' #? 'bar').a;
  a  
-----
 foo
(1 row)

SELECT ('foo' #? 'bar').b;
  b  
-----
 bar
(1 row)

ROLLBACK;
